### ISSUE TITLE
Validate envelopes received over the network or loaded from persisted.

### ISSUE THREAD

**Issue by jmalloc** [dogmatiq/infix#153]
The validation logic is already there, it's just not being used.

**Comment by jmalloc** [comment]
Blocked by #131, or more specifically #152

**Comment by jmalloc** [comment]
This is less important than it sounds because validation is already performed when the envelope is used to construct a parcel in `parcel.FromEnvelope()`, but these envelopes could be rejected even earlier if they're not even well-formed enough to log about, etc.

**Comment by danilvpetrov** [comment]
IDK if this is related to this issue, but `dogma/verity`  does not seem to call `dogma.ValidateMessage()` the same way as `dogma/testkit` does. Is this behaviour configurable? May be I am just missing the configuration.

**Comment by jmalloc** [comment]
Yeah, def related to this issue, the engine doesn't invoke the application-level message validation stuff at all atm.
